### ISSUE TITLE
Crash on throw expression

### ISSUE THREAD

**Issue by Nek-** [FriendsOfPHP/PHP-CS-Fixer#6648]
Hello, first, thank you so much for php-cs-fixer and the work on it.

Now I'm sorry, but I'm going to describe an issue: PHP 8+ allows to use of [exception in expression](https://wiki.php.net/rfc/throw_expression), and sometimes php-cs-fixer acts bad about it...

## Bug report

PHP CS Version:

```
PHP CS Fixer 3.9.5 Grand Awaiting by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.1.8
```

Stack trace

```
        [InvalidArgumentException]                               
        Invalid param $startIndex - not a proper block "start".  
                                                                 

      PhpCsFixer\Tokenizer\Tokens->findOppositeBlockEdge()
        in /api/tools/php-cs-fixer/vendor/friendsofphp/php-cs-fixer/src/Tokenizer/Tokens.php at line 424
      PhpCsFixer\Tokenizer\Tokens->findBlockEnd()
        in /api/tools/php-cs-fixer/vendor/friendsofphp/php-cs-fixer/src/Fixer/FunctionNotation/SingleLineThrowFixer.php at line 85
      PhpCsFixer\Fixer\FunctionNotation\SingleLineThrowFixer->applyFix()
        in /api/tools/php-cs-fixer/vendor/friendsofphp/php-cs-fixer/src/AbstractFixer.php at line 75
      PhpCsFixer\AbstractFixer->fix()
        in /api/tools/php-cs-fixer/vendor/friendsofphp/php-cs-fixer/src/Runner/Runner.php at line 173
      PhpCsFixer\Runner\Runner->fixFile()
        in /api/tools/php-cs-fixer/vendor/friendsofphp/php-cs-fixer/src/Runner/Runner.php at line 114
      PhpCsFixer\Runner\Runner->fix()
        in /api/tools/php-cs-fixer/vendor/friendsofphp/php-cs-fixer/src/Console/Command/FixCommand.php at line 307
      PhpCsFixer\Console\Command\FixCommand->execute()
        in /api/tools/php-cs-fixer/vendor/symfony/console/Command/Command.php at line 291
      [ ... ]
```

### Code snippet that reproduces the problem

```php
<?php
$var = [
    $something['foo'] ?? throw new Exception('yo')
];
```

### Fix if you have the issue

```php
<?php
$var = [
    $something['foo'] ?? throw new Exception('yo'),
];
```

**Comment by SpacePossum** [comment]
Hi and thanks for reporting!

Suggested fix is here https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6649

**Comment by keradus** [comment]
closed by #6653
